From abbotti@mev.co.uk  Fri Nov 20 09:11:49 2009
From: Ian Abbott <abbotti@mev.co.uk>
Date: Fri, 20 Nov 2009 11:32:36 +0000
Subject: Staging: comedi: serial2002: decrease stack usage
Cc: Frank Mori Hess <fmhess@users.sourceforge.net>, Ian Abbott <abbotti@mev.co.uk>, Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <1258716758-9033-2-git-send-email-abbotti@mev.co.uk>


512 bytes of stack can be saved in serial_2002_open() by modifying
'struct config_t'.  A short int suffices for the 'kind' and 'bits'
members.  (Actually, a char would suffice, but wouldn't save any more
stack than a short int.)

Signed-off-by: Ian Abbott <abbotti@mev.co.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/comedi/drivers/serial2002.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/staging/comedi/drivers/serial2002.c
+++ b/drivers/staging/comedi/drivers/serial2002.c
@@ -404,8 +404,8 @@ static void serial_2002_open(struct come
 	} else {
 		struct config_t {
 
-			int kind;
-			int bits;
+			short int kind;
+			short int bits;
 			int min;
 			int max;
 		};
